app/models/spree/user_decorator.rb

```diff
@@ -4,7 +4,7 @@
   devise :omniauthable
 
   def apply_omniauth(omniauth)
-    if %w(facebook google_oauth2).include? omniauth['provider']
+    if omniauth.fetch('info', {})['email'].present?
```

Comment:
Nah, don't bother. I double checked and this extension should maintain support back to Ruby 2.1 so it can support Solidus >= 1.0. No reason to add a hack when this works just fine. Thanks for looking into that though. 😃